src/webhook/handlers/__tests__/groupHandler.test.js

```diff
@@ -0,0 +1,460 @@
+          "paused": 0,
+          "waiting": 0,
+        }
+      `);
+      expect(await expiredJobQueue.getJobCounts()).toMatchInlineSnapshot(`
+        Object {
+          "active": 0,
+          "completed": 0,
+          "delayed": 0,
+          "failed": 0,
+          "paused": 0,
+          "waiting": 0,
+        }
+      `);
+      expect(processGroupEvent).toHaveBeenCalledTimes(1);
+      expect(lineClient.post).toHaveBeenCalledTimes(1);
+      done();
+    }
+  });
+});
+
+it('should jobQueue failed with TimeoutError and should not add job to expiredQueue', done => {
+  const jobId = 'testId';
+  const { type, replyToken, ...otherFields } = jobData.textMessage;
+  const param = {
+    req: {},
+    replyToken,
+    type,
+    groupId: otherFields.source.groupId,
+    otherFields: { ...otherFields },
+  };
+  processGroupEvent.mockImplementationOnce(() =>
+    Promise.reject(new TimeoutError('mock processGroupEvent timeout error'))
+  );
+  const gh = new GroupHandler(jobQueue, expiredJobQueue, 1);
+  gh.addJob(param, { jobId });
+
+  jobQueue.on('failed', async (job, e) => {
+    // console.log('jobQueue.failed');
+    expect(e instanceof TimeoutError).toBe(true);
+    expect(e).toMatchInlineSnapshot(
+      `[Error: mock processGroupEvent timeout error]`
+    );
+  });
+  jobQueue.on('drained', async () => {
+    if ((await isQueueIdle(jobQueue)) && (await isQueueIdle(expiredJobQueue))) {
+      // console.log('test 3 done');
+      expect(await jobQueue.getJobCounts()).toMatchInlineSnapshot(`
+        Object {
+          "active": 0,
+          "completed": 0,
+          "delayed": 0,
+          "failed": 1,
+          "paused": 0,
+          "waiting": 0,
+        }
+      `);
+      expect(await expiredJobQueue.getJobCounts()).toMatchInlineSnapshot(`
+        Object {
+          "active": 0,
+          "completed": 0,
+          "delayed": 0,
+          "failed": 0,
+          "paused": 0,
+          "waiting": 0,
+        }
+      `);
+      expect(processGroupEvent).toHaveBeenCalledTimes(1);
+      expect(lineClient.post).toHaveBeenCalledTimes(0);
+      done();
+    }
+  });
+});
+
+it('should jobQueue failed with TimeoutError and add job to expiredQueue', done => {
+  const jobId = 'testId';
+  const { type, replyToken, ...otherFields } = jobData.expiredTextMessage;
+  const param = {
+    req: {},
+    replyToken,
+    type,
+    groupId: otherFields.source.groupId,
+    otherFields: { ...otherFields },
+  };
+  processGroupEvent.mockImplementationOnce(() =>
+    Promise.reject(new TimeoutError('mock processGroupEvent timeout error'))
+  );
+  const gh = new GroupHandler(jobQueue, expiredJobQueue, 1);
+  gh.addJob(param, { jobId });
+
+  jobQueue.on('failed', async (job, e) => {
+    // console.log('jobQueue.failed');
+    expect(job.id).toBe(jobId);
+    expect(e instanceof TimeoutError).toBe(true);
+    expect(e).toMatchInlineSnapshot(`[Error: Event expired]`);
+  });
+
+  // in real case this will not happen
```

Comment:
It seems that it should happen in real case as well.

When a job is added to `expiredJobQueue` in `processJob`, it must have an expired timestamp.
By the time the job is executed by `expiredJobQueue`, `processGroupEvent` will be invoked, and `isEventExpired()` inside will still throw a `TimeoutError`.

Reply:
Sorry, the comment may come from copy and paste and I forget to remove it.